hw/lint/parse-lint-report.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Parses lint report and dump filtered messages in hjson format.
+"""
+import argparse
+import re
+import mistletoe
+import datetime
+import hjson
+
+from pathlib import Path
+
+def get_results(resdir):
+    """
+    Parse report and corresponding logfiles and extract error, warning
+    and info messages for each IP present in the result folder
+    """
+    results = {"tool": "ascentlint",
+               "errors" : [],
+               "warnings" : [],
+               "lint_errors" : [],
+               "lint_warnings" : [],
+               "lint_infos" : []}
+    try:
+        # check the log file for flow errors and warnings
+        with Path(resdir).joinpath('ascentlint.log').open() as f:
+            full_file = f.read()
+            f.close()
+            results["errors"]  += re.findall(re.compile("^FlexNet Licensing error.*", re.MULTILINE),
```

Comment:
Only use `re.compile()` if you want to re-use the regex (and assign it to a variable for that purpose). Otherwise, simply use `r` strings.

Reply:
indeed, will fix.

Reply:
I thought I have to use compile in order to pass `re.MULTILINE`, but it looks like I can just do that using the `flags` argument as well...